src/cargo/core/workspace.rs

```diff
@@ -143,6 +147,11 @@ impl<'cfg> Workspace<'cfg> {
         ws.target_dir = config.target_dir()?;
         ws.root_manifest = ws.find_root(manifest_path)?;
         ws.find_members()?;
+        ws.resolve_behavior = match ws.root_maybe() {
+            MaybePackage::Package(p) => p.manifest().resolve_behavior(),
+            MaybePackage::Virtual(vm) => vm.resolve_behavior(),
+        }
+        .unwrap_or(ResolveBehavior::V1);
```

Comment:
The default here is written down in a few places, but could `Option` be plumbed through further perhaps? Ideally we'd only have one location where `None` is resolved to the actual resolve version, and that'd also ideally be at a location where we have contextual information like the workspace/package/etc to look at `edition` values.

Reply:
Yep, I'm not sure why I didn't do that initially.